src/ObjLoading/Game/T6/BSP/BSP.h

```diff
@@ -0,0 +1,186 @@
+#pragma once
+
+#include "Game/T6/T6.h"
+#include "Utils/Logging/Log.h"
+
+#include <cassert>
+#include <memory>
+#include <string>
+#include <vector>
+using namespace T6;
+
+namespace BSP
+{
+    enum BSPMaterialType
+    {
+        MATERIAL_TYPE_COLOUR,
+        MATERIAL_TYPE_TEXTURE,
+        MATERIAL_TYPE_EMPTY
+    };
+
+    struct BSPVertex
+    {
+        vec3_t pos;
+        vec4_t color;
+        vec2_t texCoord;
+        vec3_t normal;
+        vec3_t tangent;
+    };
+
+    struct BSPMaterial
+    {
+        BSPMaterialType materialType;
+        std::string materialName;
+    };
+
+    struct BSPSurface
+    {
+        BSPMaterial material;
+        int triCount;
+        int indexOfFirstVertex;
+        int indexOfFirstIndex;
+    };
+
+    struct BSPWorld
+    {
+        std::vector<BSPSurface> surfaces;
+        std::vector<BSPVertex> vertices;
+        std::vector<uint16_t> indices;
+    };
+
+    struct BSPData
+    {
+        std::string name;
+        std::string bspName;
+
+        BSPWorld gfxWorld;
+        BSPWorld colWorld;
+    };
+
+    // BSPGameConstants:
+    // These values are hardcoded ingame and will break the map if they are changed
+    namespace BSPGameConstants
+    {
+        constexpr unsigned int MAX_COLLISION_VERTS = UINT16_MAX;
```

Comment:
nit: when there is a c++ way of doing things, i often prefer it over the C way 😅 

Numeric limits in c++ are defined via [`std::numeric_limits`](https://en.cppreference.com/w/cpp/types/numeric_limits.html).
While it may have the small benefit of always being typed correctly by default, it is mostly a personal taste thing.

adjusted in: 27c3c7dccb1069d7a7e9d26645261cb5d9c0482d